src/GCDWheel.cs

```diff
@@ -269,51 +341,621 @@ public unsafe class GCDWheel {
+                string abilityNameOutput = shortCastCachedSpellName;
+                if (queuedAbilityName != " " && conf.CastBarShowQueuedSpell)
+                    abilityNameOutput += " -> " + queuedAbilityName;
+                if (!string.IsNullOrEmpty(abilityNameOutput))
+                    DrawBarText(ui, abilityNameOutput);
+            }
+
+        }
+
+        public void DrawCastBar (PluginUI ui) {
+            float gcdTotal = DataStore.Action->TotalGCD;
+            float castTotal = DataStore.Action->TotalCastTime;
+            float castElapsed = DataStore.Action->ElapsedCastTime;
+            float castbarProgress = castElapsed / castTotal;
+            float castbarEnd = 1f;
+            float slidecastOffset = 0.5f;
+            float slidecastStart = Math.Max((castTotal - slidecastOffset) / castTotal, 0f);
+            float slidecastEnd = castbarEnd;
+
+            // handle short casts
+            if (gcdTotal > castTotal) {
+                castbarEnd = castTotal / gcdTotal;
+                slidecastStart = Math.Max((castTotal - slidecastOffset) / gcdTotal, 0f);
+                slidecastEnd = conf.SlideCastFullBar ? 1f : castbarEnd;
+            }
+
+            DrawBarElements(ui, true, gcdTotal > castTotal, castbarProgress * castbarEnd, slidecastStart, slidecastEnd);
+
+            // Text
+            // reset the queued name when we start to cast.
+            if (castbarProgress <= 0.25f)
+                queuedAbilityName = " ";
+            if (!string.IsNullOrEmpty(GetCastbarContents())) {
+                if (castbarEnd - castbarProgress <= 0.01f && gcdTotal > castTotal) {
+                    shortCastFinished = true;
+                    shortCastCachedSpellName = GetCastbarContents();
+                }
+                string abilityNameOutput = GetCastbarContents();
+                if (conf.castTimePosition == 0 && conf.CastTimeEnabled)
+                    abilityNameOutput += " (" + remainingCastTimeString + ")";
+                if (queuedAbilityName != " " && conf.CastBarShowQueuedSpell)
+                    abilityNameOutput += " -> " + queuedAbilityName;
+                    
+                DrawBarText(ui, abilityNameOutput);
+            }
+        }
+
+        private void DrawBarText(PluginUI ui, string abilityName){
+            int barWidth = (int)(ui.w_size.X * conf.BarWidthRatio);
+            string combinedText = abilityName + remainingCastTimeString + "!)/|";
+            Vector2 spellNamePos = new(ui.w_cent.X - ((float)barWidth / 2.05f), ui.w_cent.Y);
+            Vector2 spellTimePos = new(ui.w_cent.X + ((float)barWidth / 2.05f), ui.w_cent.Y);
+
+            // probably better to move the check for conf.EnableCastText to 
+            // happen before we query the text contents from the game,
+            // but I put it here for now so that we'd be "ready" if someone
+            // was playing with the checkbox while casting (e.g. there'd be stuff
```

Comment:
I think this is less relevant now that we are using ATKStage.  I doubt the performance hit is anything at all, and the ATKStage approach doesn't require any setup, since the game will always have this data available for us.